### ISSUE TITLE
Support collapsed subprocesses in C8

### ISSUE THREAD

**Issue by CatalinaMoisuc** [camunda/camunda-modeler#4020]
_Implementation task for https://github.com/camunda/product-hub/issues/294._

### Problem you would like to solve

Camunda 8 (target version pending) is going to support collapsed sub-processes. Our tool should reflect that.

### Proposed solution

Enable support for collapsed subprocesses in C8:

* [x] Validation allows collapsed sub-processes for `8.4`
* [x] Validation disallows collapsed sub-process with aligned message `Only supported in 8.4 and newer` for older Camunda versions.

Follow-up in Web Modeler:

* [ ] Allow collapsed sub-process modeling (already allowed in Desktop Modeler) - https://github.com/camunda/web-modeler/issues/2969

### Alternatives considered

None.

### Additional context

We currently support modeling already, but mark the sub-process as "unsupported", using an outdated message:

![image](https://github.com/camunda/camunda-modeler/assets/58601/aeccb4e8-a828-4f57-a4a4-13a82ffc1cce)

**Comment by nikku** [comment]
@CatalinaMoisuc I updated the issue. From what I see we still lack the target version.

**Comment by nikku** [comment]
Moving to `ready` in that case.

**Comment by nikku** [comment]
I updated the description, also flagging the follow-up needed in web modeler.

**Comment by barmac** [comment]
I will look into this tomorrow.

**Comment by barmac** [comment]
This will be completed via:
- https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/150
- https://github.com/camunda/linting/pull/95

**Comment by HanselIdes** [comment]
This looks great for Modeler and Operate!

@crobbins215, one requirement may have been missed
In the PDP epic https://github.com/camunda/product-hub/issues/294#issuecomment-1835701940 I added a must-have requirement to explain the limitations in Optimize with collapsed subprocesses. I also added a user story: "As a Developer, I am warned that analysis of the collapsed sub-process is not supported in Optimize", and that hasn't been removed. This _could_ also be done through documentation, but it's your call.
